sktime/param_est/base.py

```diff
@@ -180,196 +193,290 @@ def fit(self, X):
         # if fit is called, estimator is reset, including fitted state
         self.reset()
 
-        # check and convert X/y
-        X_inner = self._check_X(X=X)
+        # Check and convert X and y according to estimator requirements
+        X_inner, y_inner = self._check_X_y(X=X, y=y)
 
         # set internal X to the new X
-        self._update_X(X_inner)
+        self._update_y_X(y_inner, X_inner)
 
         # checks and conversions complete, pass to inner fit
         #####################################################
-        self._fit(X=X_inner)
+        self._fit(X=X_inner, y=y_inner)
```

Comment:
I think we need to be careful here!

Imagine current estimators in a 3rd party package where `_fit` has only `X` as argument. All those estimators would now break, because you are passing an argument that is not available!

We need to ensure that we do not pass `y` when it is not available in the inner method as an argument. I will refactor some code that already does this and link the PR here so you can depend on it.

We should also test somewhere that inner estimators without `y` are still supported.

One way is to just not change all the estimators that do not use `y`.